### PR TITLE
feat(datepicker): vue2 style and keyboard navigation enhancement

### PR DESCRIPTION
**Pull request by iropolo**
# Vue2 Datepicker style and keyboard navigation enhancement 

Jira Ticket: [DLT-1392](https://dialpad.atlassian.net/browse/DLT-1392)

HTML structure and style comes from this PR #35 

In addition there is a keyboard navigation improvement with the following instructions:

- if focus is on last day of month, pressing `→` should go to first day of next month
- if focus is on first day of month, pressing `←` should go to last day of previous month 
- if focus is on a day of first week, pressing `↑` should go to corresponding day (e.g. Thursday) of previous month's week. 
- if focus is on a day of last week, pressing `↓` should go to corresponding day (e.g. Thursday) of next month's week.

[DLT-1392]: https://dialpad.atlassian.net/browse/DLT-1392?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by iropolo** [comment]
Visual tests passes, it is okay, we do not expect to have visual changes.

**Comment by iropolo** [comment]
Tests updated with new structure and new functionalities.

**Comment by iropolo** [comment]
Vue3 update, should go tomorrow.

File: packages/dialtone-css/lib/build/less/components/datepicker.less [hunk 1]
```diff
@@ -12,135 +12,79 @@
...
   padding: var(--dt-space-500);
+  box-sizing: border-box;
 
-  p {
-    display: flex;
-    color: var(--dt-color-foreground-secondary);
-    font-size: var(--dt-font-size-100);
-    text-transform: uppercase;
+  &__hd {
+    padding: 0 4px;
```
Comment (ninamarina) [inline packages/dialtone-css/lib/build/less/components/datepicker.less]:
```suggestion
    padding: 0 var(--dt-space-300);
```

File: packages/dialtone-css/lib/build/less/components/datepicker.less [hunk 2]
```diff
@@ -12,135 +12,79 @@
...
 
-    > div {
-      display: flex;
-      gap: var(--dt-space-200);
-    }
+  &__month-year-title {
+    line-height: var(--dt-font-line-height-300);
+    font-size: var(--dt-font-size-100);
+    font-weight: var(--dt-font-weight-normal);
+    text-transform: uppercase !important;
```
Comment (ninamarina) [inline packages/dialtone-css/lib/build/less/components/datepicker.less]: You can remove this `!important`

**Review by braddialpad — APPROVED** [review]
- keyboard functionality looks good
- screenreader looks good
- semantic elements look good
- styles look good

Not seeing any problems here nice work!

**Comment by yorbi-dp** [comment]
Hey! I believe this is an unexpected outcome:
https://dialtone.dialpad.com/deploy-previews/pr-95/components/datepicker.html
Screenshot 2024-01-16 at 11 51 14 AM

**Comment by braddialpad** [comment]
> Hey! I believe this is an unexpected outcome:

That certainly is haha. Looks fine on storybook.

**Comment by iropolo** [comment]
Huh, thats the dialtone-documentation project, lemme look at it

**Comment by iropolo** [comment]
Hmm... looks well on vue3 PR #98 
https://dialtone.dialpad.com/deploy-previews/pr-98/components/datepicker.html

dialtone-documentation is using vue3. 
this PR will not have it ok. (this is vue2 PR).

It will be fix after the merge & deploy.
There is nothing to do now.
@yorbi-dp

**Comment by braddialpad** [comment]
oh yeah it's just like that because in this PR vue 3 hasn't been done yet. Makes sense.

**Pull request merged**